boa/src/builtins/array/mod.rs

```diff
@@ -292,25 +291,32 @@ impl Array {
     /// Returns a Boolean valued property that if `true` indicates that
     /// an object should be flattened to its array elements
     /// by `Array.prototype.concat`.
-    fn is_concat_spreadable(this: &GcObject, context: &mut Context) -> Result<bool> {
+    fn is_concat_spreadable(this: &Value, context: &mut Context) -> Result<bool> {
         // 1. If Type(O) is not Object, return false.
+        if !this.is_object() {
+            return Ok(false);
+        }
         // 2. Let spreadable be ? Get(O, @@isConcatSpreadable).
-        let spreadable = this.get(WellKnownSymbols::is_concat_spreadable(), context)?;
+        let spreadable = this.get_field(WellKnownSymbols::is_concat_spreadable(), context)?;
+
         // 3. If spreadable is not undefined, return ! ToBoolean(spreadable).
         if !spreadable.is_undefined() {
             return Ok(spreadable.to_boolean());
         }
         // 4. Return ? IsArray(O).
-        Ok(this.is_array())
+        match this.as_object() {
+            Some(obj) => Ok(obj.is_array()),
+            _ => Ok(false),
+        }
     }
 
     /// `LengthOfArrayLike ( obj )`
     ///
     /// The abstract operation LengthOfArrayLike takes argument obj.
     /// It returns the value of the "length" property of an array-like object (as a non-negative integer).
 
-    fn length_of_array_like(this: &GcObject, context: &mut Context) -> Result<usize> {
-        this.get("length", context)?.to_length(context)
+    fn length_of_array_like(this: &Value, context: &mut Context) -> Result<usize> {
+        this.get_field("length", context)?.to_length(context)
```

Comment:
we already have an implementation of `length_of_array_like` in `GcObject` as a method.